test/hotspot/jtreg/vmTestbase/nsk/jdi/ClassPrepareEvent/thread/thread001.java

```diff
@@ -225,11 +231,17 @@ public void run() {
                                           }
 
                                           // Check that all expected ClassPrepareEvent are received
-                                          eventsReceived = true;
-                                          for (int i = 0; i < checkedThreads.length; i++) {
-                                               if (checkedThreads[i][2] == "0") {
-                                                    eventsReceived = false;
-                                               }
+                                          if (!eventsReceived) {
+                                              eventsReceived = true;
+                                              for (int i = 0; i < checkedThreads.length; i++) {
+                                                  if (checkedThreads[i][2] == "0") {
```

Comment:
This if-statement could use a comment to explain the logic.

Reply:
Done.